drivers/flash/flash_hp_ra.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_FLASH_RA_HP_H_
+#define ZEPHYR_DRIVERS_FLASH_RA_HP_H_
+
+#include <zephyr/drivers/flash.h>
+#include <instances/r_flash_hp.h>
+#include <api/r_flash_api.h>
+#include <zephyr/drivers/flash/ra_flash_api_extensions.h>
+
+#define UNIT_CODE_FLASH_PROGRAMMING 128
+
+enum flash_region {
+	CODE_FLASH,
+	DATA_FLASH,
+};
+
+typedef void (*irq_config_func_t)(const struct device *dev);
+
+struct flash_hp_ra_controller {
+	struct st_flash_hp_instance_ctrl flash_ctrl;
+	struct k_sem ctrl_sem;
+	struct st_flash_cfg fsp_config;
+};
+
+struct flash_hp_ra_controller_config {
+	irq_config_func_t irq_config;
+};
+
+struct flash_hp_ra_data {
+	struct flash_hp_ra_controller *controller;
+	enum flash_region FlashRegion;
+	off_t area_address;
+	off_t area_size;
+	off_t erase_block_size;
```

Comment:
you probably want to use `uint32_t` or `uint64_t` directly here. The `off_t` is signed so that you could offset in the negative direction or use it for in-band error reporting; in your case these will always be positive, so `off_t` does not make sense here.

Reply:
I update it into uint32_t. Thanks